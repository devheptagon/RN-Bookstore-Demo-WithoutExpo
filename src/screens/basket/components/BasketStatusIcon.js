import React from "react";
import PropTypes from "prop-types";
import Icon from "react-native-vector-icons/SimpleLineIcons";
import { StyleSheet, TouchableHighlight } from "react-native";
import { Badge } from "react-native-elements";
import FlexView from "../../components/RegularView";
import Theme from "../../../constants/Theme";
import { FlexDirections, IconNames } from "../../../constants/Enums";

const BasketStatusIcon = props => (
  <TouchableHighlight onPress={props.onPress}>
    <FlexView flexDirection={FlexDirections.Row}>
      {props.itemCount ? (
        <Badge
          value={props.itemCount}
          status="error"
          badgeStyle={styles.Badge}
        />
      ) : null}
      <Icon name={IconNames.BasketStatus} size={25} style={styles.BasketIcon} />
    </FlexView>
  </TouchableHighlight>
);

const styles = StyleSheet.create({
  BasketIcon: {
    color: Theme.HeaderBar.IconColor,
    marginLeft: 5,
    zIndex: 1
  },
  Badge: {
    marginRight: -40,
    marginTop: 3,
    zIndex: 2
  }
});

BasketStatusIcon.propTypes = {
  itemCount: PropTypes.number
};

export default BasketStatusIcon;
