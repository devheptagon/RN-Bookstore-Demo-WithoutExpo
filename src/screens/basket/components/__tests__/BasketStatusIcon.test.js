import React from 'react';
import BasketStatusIcon from '../BasketStatusIcon';
import renderer from 'react-test-renderer';

test('BasketStatusIcon renders correctly', () => {
  const tree = renderer.create(<BasketStatusIcon />).toJSON();
  expect(tree).toMatchSnapshot();
});