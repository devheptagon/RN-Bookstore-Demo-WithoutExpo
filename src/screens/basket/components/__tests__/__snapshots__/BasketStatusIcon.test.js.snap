// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BasketStatusIcon renders correctly 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={null}
>
  <View
    flexDirection="row"
    style={Object {}}
  >
    <Text />
  </View>
</View>
`;
